submissions/Complexity Corner - The Mall/Complexity Corner - The Mall.md

```diff
@@ -0,0 +1,105 @@
+
+[bot-based mall example]
+
+[second example from community?]
+
+## **How are the machine outputs controlled and stored?**
+
+The intuitive way to collect the output from your machines is by adding chests. Bigger chests hold more items and passive provider (red) chests will let bots pickup from the mall too. However, you rarely need more than a stack or two of anything at a time. Hence, it would be preferable to prevent overproducing items at your mall and draining your raw materials.
+
+The easiest way to control machine outputs is the "red x" inventory limitation option for chests. You can use it to limit mall output chests to 1-2 stacks each. Yet in the early game, sometimes even one stack is too big. In that case why not add a single transport belt pointing towards the output inserter? The belt in this orientation will hold only five (todo* verify) items, plus you'll have an extra handful in the inserter and machine. If you want to restrict outputs to even less, you also have the option of have an inserter place simply onto the ground or having no inserter at all.
+
+[low-tech output limiting methods: no output (0+n) , 1-belt output (5+n) , limited chest output (1 stack + n)]
+
+The more precise way to control your mall output is by using circuit conditions and no combinators are needed for this! The first setup idea here is to wire each output inserter to its own output chest. The wire reads the chest contents. All you have to do is set the inserter enable condition to "*everything* < X" with X being the output count you want. This condition is easy to copy-paste and will work for any output chest. This setup is nice also because you don't have to limit the chest and so you can dump extras of the output item back in it easily, hence making your mall output chests function also as specialized storage chests. 
+
+Speaking of specialized storage, for a bot-based mall you can also use logistic storage (yellow) chests to collect outputs and set their filters to the machine's output items. This setup will make bots return trashed items to the mall storage and prevent overproduction accumulating across your logistic storage. When you combine the storage chest setup with the circuit wire setup, you get very precise control of item counts in your logistic network. Meanwhile, there is also a zero-wire version of this combination: You can just use the output inserter's logistic network connection tab and set "*output item* < X" there. This alternative cannot use the "everything" signal, though.
```

Comment:
also valid with storage chests with bots later on, when you convert: so that when you deconstruct your yellow belts, they can be inserted back for red belts production and no crafting new ones waiting for this backlog to be upgraded :)

Reply:
I like the yellow belt example to explain the ideas here but i did not understand which sentence you replied to here.

Reply:
"Speaking of specialized storage, for a bot-based mall you can also use logistic storage (yellow) chests to collect outputs and set their filters to the machine's output items."
my comment is to complete this idea, to add "recycling" behavior in addition to the indented crafting behavior